### PR TITLE
Add InClusterconfig for sparkctl if no kubeconfig provided

### PR DESCRIPTION
**Pull request by sarjeet2013**
Fixes https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/419

### CODE REVIEW

File: sparkctl/cmd/client.go [hunk 1]
```diff
@@ -24,9 +24,22 @@ import (
 
 	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
 	crdclientset "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/client/clientset/versioned"
+	"os"
 )
 
 func buildConfig(kubeConfig string) (*rest.Config, error) {
+
+	// Check if kubeConfig exist
```
Comment (liyinan926) [inline sparkctl/cmd/client.go]: The right fix I think should be to change `getKubeConfigPath` in `root.go` to return an empty string when the default path does not exist. `BuildConfigFromFlags` will use `InClusterConfig` if the input path is empty.

Comment (sarjeet2013) [reply sparkctl/cmd/client.go]: Right, The `getKubeConfigPath` return the default path if `KUBECONFIG` env is not present. In that case, we can check if the default path exist as well, and return "" if it doesn't exist.

I'll update the PR

**Comment by sarjeet2013** [comment]
@liyinan926 I had it modified as you suggested, but now this is getting the warning log before trying incluster config

sh-4.2# sparkctl list
ERROR: logging before flag.Parse: W0221 19:48:37.844107    1041 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
+------+-------+----------------+-----------------+
| NAME | STATE | SUBMISSION AGE | TERMINATION AGE |
+------+-------+----------------+-----------------+
+------+-------+----------------+-----------------+
This warning is from `BuildConfigFromFlags` where kubeconfig is provided “”

Should we stick to original PR?

File: sparkctl/cmd/client.go [hunk 2]
```diff
@@ -24,9 +24,22 @@ import (
 
 	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
 	crdclientset "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/client/clientset/versioned"
+	"os"
 )
 
 func buildConfig(kubeConfig string) (*rest.Config, error) {
+
```
Comment (liyinan926) [inline sparkctl/cmd/client.go]: This empty line is not needed.

File: sparkctl/cmd/client.go [hunk 3]
```diff
@@ -24,9 +24,22 @@ import (
 
 	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
 	crdclientset "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/client/clientset/versioned"
+	"os"
```
Comment (liyinan926) [inline sparkctl/cmd/client.go]: This should be at the top of the import list with other built-in packages.

Comment (sarjeet2013) [reply sparkctl/cmd/client.go]: Fixed

File: sparkctl/cmd/client.go (hunk 2 above)
Comment (sarjeet2013) [reply sparkctl/cmd/client.go]: Done, Removed

**Comment by sarjeet2013** [comment]
@liyinan926 Updated, PTAL. Thanks for the review.

File: sparkctl/cmd/client.go [hunk 4]
```diff
@@ -21,12 +21,24 @@ import (
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
+	"os"
```
Comment (liyinan926) [inline sparkctl/cmd/client.go]: Sorry I meant this should be in the group of golang built-in packages.

Comment (sarjeet2013) [reply sparkctl/cmd/client.go]: You mean, "os" at the first line in import, then an empty line, and then the rest of the import packages?

Comment (sarjeet2013) [reply sparkctl/cmd/client.go]:
```import (
	"os"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	clientset "k8s.io/client-go/kubernetes"
	"k8s.io/client-go/rest"
	"k8s.io/client-go/tools/clientcmd"

	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
	crdclientset "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/client/clientset/versioned"
)```

Comment (liyinan926) [reply sparkctl/cmd/client.go]: Yes

Comment (sarjeet2013) [reply sparkctl/cmd/client.go]: :) Done

**Pull request merged**